utils/data.hpp

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include "shapes/Shapes.hpp"
+#include "Canvas.hpp"
+#include "exceptions.hpp"
+#include <QString>
+#include <QChar>
+#include <QMap>
+#include <QColor>
+#include <QFont>
+#include <QFile>
+#include <QFileDialog>
+#include <QTextStream>
+
+namespace jbrush {
+// Directory starts at this string
+const QString STARTING_DIRECTORY = "/";
+
+// Data class member functions expect these strings as identifiers for the shape that is being read in
+const QString LINE = "Line";
+const QString POLYLINE = "Polyline";
+const QString POLYGON = "Polygon";
+const QString RECTANGLE = "Rectangle";
+const QString SQUARE = "Square";
+const QString ELLIPSE = "Ellipse";
+const QString CIRCLE = "Circle";
+const QString TEXT = "Text";
+const QStringList TYPE_LIST = {LINE, POLYLINE, POLYGON, RECTANGLE, SQUARE, ELLIPSE, CIRCLE, TEXT };
+
+// The parser expects this character at the end of descriptive titles for data
+const QChar TITLE_INTERRUPTOR = ' ';
+// The parser expects this character to interrupt each piece of data in the line of text that gives the shape's dimensions
+const QChar DIMENSION_INTERRUPTOR = ',';
+
+// Pairs of parallel lists indirectly associate strings with corresponding values
+const QStringList PEN_STYLE_STR = {
```

Comment:
If any of these can be changed to standard arrays, I don't think we need the extra functionality from Qt

Reply:
I have to use `QList.idexOf()` in the `StringToEnum()` method, though.  I guess I could use arrays and write my own templated `indexOf()` method for that, but I just figured it'd be a waste of time to rewrite that code myself.  Still think it'd be better to switch to arrays or should I leave it as is?